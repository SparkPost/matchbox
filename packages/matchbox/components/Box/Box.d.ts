export default Box;
declare const Box: any;
