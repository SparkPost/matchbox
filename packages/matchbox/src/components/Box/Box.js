import styled from 'styled-components';
import { border, color, flexbox, grid, layout, position, shadow, space, typography, compose } from 'styled-system';
import propTypes from '@styled-system/prop-types';

const system = compose(
  border,
  color,
  flexbox,
  grid,
  layout,
  position,
  shadow,
  space,
  typography
);

const truncate = (props) => {
  if (props.truncate) {
    return {
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap'
    };
  }
};

const Box = styled('div')`
  ${system}
  ${truncate}
`;

Box.propTypes = {
  ...propTypes.border,
  ...propTypes.color,
  ...propTypes.flexbox,
  ...propTypes.grid,
  ...propTypes.layout,
  ...propTypes.position,
  ...propTypes.shadow,
  ...propTypes.space,
  ...propTypes.typography
};

export default Box;
