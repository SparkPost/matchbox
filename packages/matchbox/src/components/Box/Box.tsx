import React from 'react';
import styled from 'styled-components';
import {
  border,
  BorderProps,
  color,
  ColorProps,
  flexbox,
  FlexboxProps,
  grid,
  GridProps,
  layout,
  LayoutProps,
  position,
  PositionProps,
  shadow,
  ShadowProps,
  space,
  SpaceProps,
  typography,
  TypographyProps,
  compose,
} from 'styled-system';

type BoxProps = BorderProps &
  ColorProps &
  FlexboxProps &
  GridProps &
  LayoutProps &
  PositionProps &
  ShadowProps &
  SpaceProps &
  TypographyProps & {
    children: React.ReactNode;
  };

const system = compose(border, color, flexbox, grid, layout, position, shadow, space, typography);

const truncate = props => {
  if (props.truncate) {
    return {
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap',
    };
  }
};

const Box = styled.div<BoxProps>`
  ${system}
  ${truncate}
`;

Box.displayName = 'Box';
export default Box;
