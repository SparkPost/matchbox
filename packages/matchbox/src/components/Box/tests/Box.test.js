import React from 'react';
import Box from '../Box';
import 'jest-styled-components';

describe('Box', () => {
  it('it should render correctly', () => {
    const wrapper = global.renderStyled(
      <Box
        py="400"
        color="gray.100"
        bg={['blue.200', 'blue.500', 'blue.800']}
        borderRadius="pill"
        border="400"
        fontSize="200"
        lineHeight="200"
        fontFamily="monospace"
      >
        Just a Box
      </Box>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('it should truncate', () => {
    const wrapper = global.renderStyled(<Box truncate>Just a truncated Box</Box>);
    expect(wrapper).toMatchSnapshot();
  });
});
