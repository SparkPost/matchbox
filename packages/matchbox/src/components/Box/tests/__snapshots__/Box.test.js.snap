// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box it should render correctly 1`] = `
.c0 {
  padding-top: 1rem;
  padding-bottom: 1rem;
  color: #f5f8fa;
  background-color: #e8f3ff;
  border-radius: 9999em;
  border: 1px solid #c5ced6;
  font-size: 1rem;
  line-height: 1.25rem;
  font-family: 'SFMono-Regular',Monaco,Consolas,'Lucida Console',monospace;
}

@media screen and (min-width:450px) {
  .c0 {
    background-color: #78b6ff;
  }
}

@media screen and (min-width:720px) {
  .c0 {
    background-color: #0060cc;
  }
}

<div
  class="c0"
  color="gray.100"
  font-family="monospace"
  font-size="200"
>
  Just a Box
</div>
`;

exports[`Box it should truncate 1`] = `
.c0 {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
  class="c0"
>
  Just a truncated Box
</div>
`;
